hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/ServicesState.java

```diff
@@ -640,7 +661,100 @@ boolean shouldMigrateAccountsToDisk() {
+            merkleDbBackedMap = migrateVirtualMap(contractStorageMap, merkleDbBackedMap);
+            state.setChild(StateChildIndices.CONTRACT_STORAGE, merkleDbBackedMap);
+        }
+
+        // virtualized accounts, if enabled
+        if (state.enableVirtualAccounts) {
+            final VirtualMap<EntityNumVirtualKey, OnDiskAccount> accountsMap =
+                    state.getChild(StateChildIndices.ACCOUNTS);
+            if (jasperDbBacked(accountsMap)) {
+                VirtualMap<EntityNumVirtualKey, OnDiskAccount> merkleDbBackedMap =
+                        virtualMapFactory.newOnDiskAccountStorage();
+                merkleDbBackedMap = migrateVirtualMap(accountsMap, merkleDbBackedMap);
+                state.setChild(StateChildIndices.ACCOUNTS, merkleDbBackedMap);
+            }
+        }
+
+        // virtualized token associations, if enabled
+        if (state.enableVirtualTokenRels) {
+            final VirtualMap<EntityNumVirtualKey, OnDiskTokenRel> tokenAssociationsMap =
+                    state.getChild(StateChildIndices.TOKEN_ASSOCIATIONS);
+            if (jasperDbBacked(tokenAssociationsMap)) {
+                VirtualMap<EntityNumVirtualKey, OnDiskTokenRel> merkleDbBackedMap =
+                        virtualMapFactory.newOnDiskTokenRels();
+                merkleDbBackedMap = migrateVirtualMap(tokenAssociationsMap, merkleDbBackedMap);
+                state.setChild(StateChildIndices.TOKEN_ASSOCIATIONS, merkleDbBackedMap);
+            }
+        }
+
+        // virtualized unique token storage, if enabled
+        if (state.enabledVirtualNft) {
+            final VirtualMap<UniqueTokenKey, UniqueTokenValue> uniqueTokensMap =
+                    state.getChild(StateChildIndices.UNIQUE_TOKENS);
+            if (jasperDbBacked(uniqueTokensMap)) {
+                VirtualMap<UniqueTokenKey, UniqueTokenValue> merkleDbBackedMap =
+                        virtualMapFactory.newVirtualizedUniqueTokenStorage();
+                merkleDbBackedMap = migrateVirtualMap(uniqueTokensMap, merkleDbBackedMap);
+                state.setChild(StateChildIndices.UNIQUE_TOKENS, merkleDbBackedMap);
+            }
+        }
+    }
+
+    private static boolean jasperDbBacked(final VirtualMap<?, ?> map) {
+        final VirtualRootNode<?, ?> virtualRootNode = map.getRight();
+        return virtualRootNode.getDataSource() instanceof VirtualDataSourceJasperDB;
+    }
+
+    private static <K extends VirtualKey<? super K>, V extends VirtualValue>
+            VirtualMap<K, V> migrateVirtualMap(
+                    final VirtualMap<K, V> source, final VirtualMap<K, V> target)
+                    throws InterruptedException {
+        final int copyTargetMapEveryPuts = 10_000;
+        final AtomicInteger count = new AtomicInteger(copyTargetMapEveryPuts);
+        final AtomicReference<VirtualMap<K, V>> targetMapRef = new AtomicReference<>(target);
+        VirtualMapMigration.extractVirtualMapData(
```

Comment:
Since `VirtualMapMigration` doesn't log any timing info out-of-the-box, it might be nice to wrap this in a [`withLoggedDuration()`](https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/utils/MiscUtils.java#L880) so we can get a feel for how long various maps are taking to migrate.

Not a blocker ofc.

Reply:
OK, let me implement this.